### ISSUE TITLE
False positive / invalid UnusedVariable, when the variable is concatenated with a lot of strings

### ISSUE THREAD

**Issue by ArtemGoutsoul** [vimeo/psalm#6297]
This code produces a notice when there are 105 string concats:
https://psalm.dev/r/e27b51a709

Code:
```php
<?php

function func(): void
{
	$year = date('Y');

	$expected = $year
	            . ""
	            . ""
	            . ""
	            . ""
// 105 concats total
	            . ""
	            . ""
	            . "";

	assert($expected === 'stuff');
}
```
Notice:
```
INFO: UnusedVariable - 5:2 - $year is never referenced or the value is not used
```

when I remove one concatenation and there are only 104 left, the error disappears:
https://psalm.dev/r/d87d6fc499

**Comment by orklah** [comment]
Most likely due to this threshold:
https://github.com/vimeo/psalm/blob/19cc4cb4ee962a9872aa1e5c61ab41ef3c4348d5/src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOpAnalyzer.php#L30

**Comment by ArtemGoutsoul** [comment]
Good to know!
For our codebase this is a rare edge case, since this type of concatenation is only used to provide a long multiline string for a test's expected value. We could relatively easily refactor it probably. For now the related error is just in the baseline.

**Comment by orklah** [comment]
I'm gonna close this as this is not really a bug. However, there are a few other places where we do that and I wonder if a new issue `Threshold reached` that is emitted when this happens wouldn't be the best in this case. It could explain that further analysis around that code may behave weirdly and that could signal the user a refactor would be beneficial...

**Comment by ArtemGoutsoul** [comment]
That new issue would be great, and this way it would let the developer make a sensible choice about how to handle it, e.g. reduce the binary op usage in question, baseline or suppress.

**Comment by orklah** [comment]
Can you create an issue here for that please?

**Comment by ArtemGoutsoul** [comment]
@orklah hopefully this makes sense: https://github.com/vimeo/psalm/issues/6379
